src/SoftRobots.Inverse/component/solver/QPInverseProblemSolver.cpp

```diff
@@ -267,6 +267,11 @@ void QPInverseProblemSolver::init()
 
     if(d_multithreading.getValue())
         sofa::simulation::MainTaskSchedulerFactory::createInRegistry()->init();
+
+    for (const auto& constraintCorrection : m_constraintsCorrections)
+    {
+        constraintCorrection->addConstraintSolver(this);
+    }
```

Comment:
Maybe for another PR, but wouldn't it be better if `QPInverseProblemSolver` inherited from `ConstraintSolverImpl` instead of `ConstraintSolver`

Reply:
Isnt'it already the case ?